### ISSUE TITLE
Specify URLs in `iso_paths` for vsphere-iso builder

### ISSUE THREAD

**Issue by gaui** [jetbrains-infra/packer-builder-vsphere#99]
It would be nice to be possible to specify URLs in the `iso_paths` field, for `vsphere-iso` builder, so it would download the ISO beforehand.

Example:

```
"iso_paths": [
	"http://releases.ubuntu.com/16.04/ubuntu-16.04.4-server-amd64.iso"
]
```

**Comment by mkuzmin** [comment]
yeah, the option in desktop builders is known for years. but working with remote hypervisor we have a set of new questions:
- what datastore should the file be uploaded to?
- delete it after a build or not?
- how to share the same files between different Packer builds?

And most important one: will you use such option in your daily production builds or only in demos?

**Comment by gaui** [comment]
Good questions.

- what datastore should the file be uploaded to?
  - _make it configurable_
- delete it after a build or not?
  - _make it configurable_
- how to share the same files between different Packer builds?
  - _I'm not sure. Maybe introduce some kind of a global registry where SHA of the files downloaded are stored? Or search by filename in vSphere datastores?_
- will you use such option in your daily production builds or only in demos?
  - production builds... although we have a test-staging-production environment, where images are tested

**Comment by VladRassokhin** [comment]
I'd say that most important question is how often you need to change/upload new file. From our perspective that happens quite rarely and complicating packer builder to support some cache is not viable option.
I would recommend uploading iso files manually, by script or using [Terraform `vsphere_file` resource](https://www.terraform.io/docs/providers/vsphere/r/file.html).

Another option to investigate: in previous versions of VSphere it was possible to passthrough local cdrom to virtual machine from Windows VSphere Console. Maybe same could be somehow done using API.

**Comment by gaui** [comment]
Regarding Terraform, I don't think it's very sufficient to upload the file with Terraform, then creating the image with Packer, then provisioning the machine again with Terraform.

I agree with the cache option, it would be out of the nature of Packer.

I really like the cdrom idea.

**Comment by jostmart** [comment]
Storing images in S3 bucket(s) would greatly help distributing ISO:s efficiently.
If packer could retrieve an ISO from an URL, this can be achieved.
